docs/saved_objects/saved_object_access_control_design_proposal.md

```diff
@@ -0,0 +1,216 @@
+# Proposed Saved Object Access Control with Workspaces
+
+## Introduction
+Issue #4615 proposed the "workspace" experience for users to organize their work within OpenSearch Dashboards. It described the experience for users to manages the permissions to access the saved objects they own. Based on the workspace experience proposal, we proposes this saved objects access control design in this document. 
+
+### Problem with current implementation
+
+As of v2.7, OpenSearch Dashboards supports [multi-tenancy feature](https://opensearch.org/docs/latest/security/multi-tenancy/tenant-index/) for saved objects access conrol. The existing  implementation closely couple OpenSearch Dashboards with one single OpenSearch cluster, which acts as the OpenSearch Dashboards data store as well as identity source. While the tenancy feature for OpenSearch Dashboards data access control is a coarse-grained model, one user will have access to all object in a tenant, or they have no access to any object in a teannt. It also requires administrators to centrally manage the user access to any tenant, and barely support saved objects sharing. 
+
+This document discusses the proposed data access control for OpenSearch Dashboards, which is independent from the *multi-tenancy* feature provided by the security plugin. 
+
+Glossary
+
+* **Saved object**: Any data that is managed by OpenSearch Dashboards are called saved objects, such as index patterns, visualization and dashboard definition. In a common setup, each saved object is a document stored in `.opensearch_dashboards` index. In some context, it is also called library items, in #4615
+* **Workspace**: a logical container of OpenSearch Dashboards features and saved objects
+
+## Assumptions & Scopes
+
+* Authentication is out of the scope. We can assume the users are authenticated and user profile (user name and group info) has been injected into each request
```

Comment:
While authentication is not in scope for this design - but there are plans to decouple authentication, lets add requirements on the current AuthN system expectations.

We've found internal features in the security plugin were actually being used as if they were part of a public interface, since this design depends on the security plugin accounting for these areas and making hard requirements will ensure no breaking changes and make migrating to the end-state Trinity authentication system more preditable.